### ISSUE TITLE
How to specify route name

### ISSUE THREAD

**Issue by jantajov** [ktsn/vue-auto-routing#21]
Hello, is there a way how to specify a name for a generated route? 

In the default setup, the route name is inferred (hopeI am not mistaken) from the directory structure. That's cool for smaller not deeply nested routes however it would be handy if I could specify route name in the page component itself.

E.g.: it would be taken from the page component name:
```html
// users/_id/orders/_page.vue
<script>
export default {
  name: 'user-orders'
}
</script>
```

I am sorry if I missed something obvious.

**Comment by ktsn** [comment]
We cannot specify it currently. Maybe I'll add `<route-config>` custom block and let specify the name in the block.

**Comment by chojnicki** [comment]
Any progress about this?

**Comment by jantajov** [comment]
> Any progress about this?

Meanwhile I came up with following workaround: 
```javascript
import generatedRoutes from 'vue-auto-routing'

for (let route of generatedRoutes) {
  if (route.meta && route.meta.name) {
    route.name = route.meta.name
  }
}
```

You have to specify the route name in the route-meta block and the you can override it.

Your page component would look like this: 

```
<route-meta>{"name": "contact-card-update"}</route-meta>

<template>
  <div>{{hello}}</div>
</template>

<script> 
.... 
</script>
```

**Comment by NikoGrano** [comment]
Hey, I love this router, but due my Breadcrumbs are generated based on this value... Cannot use this... Could somebody expirenced take a look into this. It could be taken from page component like

```js
data() {
    return {
         router: {
            name: "xxxx",
            label: "Xxxxx"
          }
    }
}
```

**Comment by jantajov** [comment]
@NikoGrano It looks like you are trying to specify router name and label inside component data. Does it means you need to change the route title during component life-cycle? If so, I don't think you can.

The static router configuration is generated before the vue app is build. And therefore cannot change after the app is started.

**Comment by NikoGrano** [comment]
@jantajov Yeh,I was thinking samw about 1 hour later after I posted that...

But the fact is, this need to be solved, so I can use this... I know this is little selfish, but I wish this could be implemented some day 🙄

**Comment by ktsn** [comment]
@NikoGrano It's impossible even if you define router config without vue-auto-routing. Router options cannot be dynamic.

**Comment by NikoGrano** [comment]
Like I posted [here](https://github.com/ktsn/vue-auto-routing/issues/21#issuecomment-581305980) I'm aware of that. Is there any other place we can define this value?

Declaring extra element seems little weird logic, but is it only way to achive this?

**Comment by ktsn** [comment]
Please file an issue to [vue-router](https://github.com/vuejs/vue-router) with your use case and discuss there. This topic is a bit off-topic from this thread.

**Comment by NikoGrano** [comment]
Well, this issue currently doesnt seem to progress anywhere, so solution to this should be discussed here. I just lost my motivation to try solve this. Thanks, but for nothing.

**Comment by ktsn** [comment]
This issue is for specifying `name` option for route with vue-auto-routing. What you want is actually changing route name dynamically which is not implemented on vue-router yet. Discussing an vue-auto-routing implementation based on not-yet-implemented feature on vue-router is just non-sense.

I'm happy to discuss and implement dynamic config change in vue-auto-routing after vue-router would implement it.

**Comment by NikoGrano** [comment]
> What you want is actually changing route name dynamically

There is no requirement to have it dynamically. The point is to have it. You're not willing to implement this even tough it's not dynamic?

**Comment by ktsn** [comment]
I'd like to implement this feature as there is proper motivation and I too personally need it. But I don't have enough bandwidth to do this as I have other OSSs which I want to focus on currently. If anyone would like to contribute it, it would be appreciated.

**Comment by jantajov** [comment]
@ktsn Can you please clean up this thread?

**Comment by ThaDaVos** [comment]
Any ETA on a solution? I like the idea of the <route-config> custom block and have it merged with the generated one... but why need the <route-meta> custom block if you have that?

**Comment by ktsn** [comment]
Implemented via ktsn/vue-route-generator#146
Please use `<route>` custom block to set name option.

```vue
<route>
{
  "name": "route-name"
}
</route>
```
